### PR TITLE
fix(FEC-9173): playback failed for unsupported ads.

### PR DESCRIPTION
**Pull request by Yuvalke**
### Description of the Changes

Add fallback for unsupported ads on LG.
IMA doesn’t return the src for unsupported ads on LG TV.

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment 
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/ima-engine-decorator.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // @flow
-import {BaseEngineDecorator, FakeEvent} from '@playkit-js/playkit-js';
+import {BaseEngineDecorator, FakeEvent, Html5EventType} from '@playkit-js/playkit-js';
```
Comment (yairans) [inline src/ima-engine-decorator.js]: `Html5EventType` is not exposed in playkit.js use `EventType`

**Comment by yairans** [comment]
@Yuvalke Change the title to fix(FEC-9173)....

File: src/ima-engine-decorator.js (hunk 1 above)
Comment (Yuvalke) [reply src/ima-engine-decorator.js]: Of course it is look on [Playkit.js](https://github.com/kaltura/playkit-js/blob/master/src/playkit.js#L100)

Comment (yairans) [reply src/ima-engine-decorator.js]: ok. it's new.
but in your usecase it's not correct because `AUTOPLAY_FAILED` it's not native HTML event

**Comment by kaltura-hooks** [comment]
Hi @Yuvalke,
Thank you for contributing this pull request!
Please sign the Kaltura CLA so we can review and merge your contribution.
Learn more at http://bit.ly/KalturaContrib

File: src/ima-engine-decorator.js [hunk 2]
```diff
@@ -17,7 +17,17 @@ class ImaEngineDecorator extends BaseEngineDecorator {
...
+    //handle events for fatal adError that doesnt return the video source
+    if (this._plugin.isAdFailedAndSourceChanged() && event.type === EventType.ERROR) {
+      this._plugin.setAdsFailed(false);
+      this._plugin.player.getVideoElement().src = this._plugin.getContentSrc();
+      this._plugin.player.play();
+      return event.defaultPrevented;
+    } else if (this._plugin.isAdFailedAndSourceChanged() && event.type === EventType.AUTOPLAY_FAILED) {
+      return event.defaultPrevented;
+    } else {
+      return this._plugin.isAdPlayingOnSameVideoTag() ? event.defaultPrevented : super.dispatchEvent(event);
```
Comment (yairans) [inline src/ima-engine-decorator.js]: move it to be first

File: src/ima-engine-decorator.js [hunk 3]
```diff
@@ -17,7 +17,17 @@ class ImaEngineDecorator extends BaseEngineDecorator {
   }
 
   dispatchEvent(event: FakeEvent): boolean {
-    return this._plugin.isAdsPlayingOnSameVideoTag() ? event.defaultPrevented : super.dispatchEvent(event);
+    //handle events for fatal adError that doesnt return the video source
+    if (this._plugin.isAdFailedAndSourceChanged() && event.type === EventType.ERROR) {
+      this._plugin.setAdsFailed(false);
```
Comment (yairans) [inline src/ima-engine-decorator.js]:
```suggestion
      this._plugin.setAdFailed(false);
```

File: src/ima-engine-decorator.js [hunk 4]
```diff
@@ -28,7 +43,7 @@ class ImaEngineDecorator extends BaseEngineDecorator {
    * @memberof ImaEngineDecorator
    */
   get paused(): boolean {
-    return this._plugin.isAdsPlayingOnSameVideoTag() ? true : super.paused;
+    return this._plugin.isAdPlayingOnSameVideoTag() ? true : super.paused;
```
Comment (yairans) [inline src/ima-engine-decorator.js]: `isAdPlayingOnSameVideoTag` exists?

File: src/ima.js [hunk 5]
```diff
@@ -378,6 +403,13 @@ class Ima extends BasePlugin implements IMiddlewareProvider, IAdsControllerProvi
     return this._contentDuration || this.player.config.sources.duration || 0;
   }
 
+  getContentSrc(): ?string {
```
Comment (yairans) [inline src/ima.js]:
```suggestion
  getContentSrc(): string {
```

File: src/ima.js [hunk 6]
```diff
@@ -163,6 +163,13 @@ class Ima extends BasePlugin implements IMiddlewareProvider, IAdsControllerProvi
    * @memberof Ima
    */
   _contentComplete: boolean;
+  /**
+   * Flag to know when ads failed.
```
Comment (yairans) [inline src/ima.js]:
```suggestion
   * Flag to know when ad failed.
```

**Pull request merged**